src/odemis/driver/lakeshore.py

```diff
@@ -0,0 +1,520 @@
+                logging.exception(e)
+
+        # called if shutdown flag is set and loop exits
+        logging.debug("Shutting down polling thread...")
+
+        
+
+
+class LakeshoreSimulator(object):
+    """
+    Simulates a Lakeshore 310
+    Same interface as the serial port
+    """
+
+    def __init__(self, timeout=1):
+        # we don't care about the actual parameters but timeout
+        self.timeout = timeout
+        self._output_buf = b""  # what the commands sends back to the "host computer"
+        self._input_buf = b""  # what we receive from the "host computer"
+        self._status_byte = 0
+
+        self._setpoint = 10
+        self._temperature = 50
+        self._heating = 1
+
+    def write(self, data):
+        self._input_buf += data
+        msgs = self._input_buf.split(b"\r")
+        for m in msgs[:-1]:
+            self._parseMessage(m)  # will update _output_buf
+
+        self._input_buf = msgs[-1]
+
+    def read(self, size=1):
+        ret = self._output_buf[:size]
+        self._output_buf = self._output_buf[len(ret):]
+
+        if len(ret) < size:
+            # simulate timeout
+            time.sleep(self.timeout)
+        return ret
+
+    def flush(self):
+        pass
+
+    def flushInput(self):
+        self._output_buf = b""
+
+    def close(self):
+        # using read or write will fail after that
+        del self._output_buf
+        del self._input_buf
+
+    def isOpen(self):
+        return hasattr(self, "_output_buf")
+
+    def _sendAnswer(self, ans):
+        self._output_buf += b"%s\r" % (ans,)
+
+    def _parseMessage(self, msg):
+        """
+        msg (str): the message to parse (without the \r)
+        return None: self._output_buf is updated if necessary
+        """
+        logging.debug("SIM: parsing %s", to_str_escape(msg))
+        msg = msg.strip()  # remove leading and trailing whitespace
+        msg = b"".join(msg.split())  # remove all space characters
+
+        if msg == b"*STB??":
+            self._sendAnswer(b"%d" % (self._status_byte,))
+        elif msg == b"*CLS":
+            self._status_byte = 0
+        elif msg == b"*IDN?":
+            self._sendAnswer(b"SIM,310,fake")
+        elif re.match(b'LOCK', msg):
+            pass
+        # Query setpoint
+        elif re.match(b'SETP\?', msg):
+            self._sendAnswer(b"%f" % (self._setpoint,))
+        # set setpoint
+        elif re.match(b"SETP", msg):
+            vals = msg[4:].split(',')
+            self._setpoint = float(vals[1])
+        # Query heating range
+        elif re.match(b'RANGE\?', msg):
+            self._sendAnswer(b"%d" % (self._heating,))
+        # set heating range
+        elif re.match(b"RANGE", msg):
+            vals = msg[5:].split(',')
+            self._heating = int(vals[1])
+        # Query temperature
+        elif re.match(b'SRDG\?', msg):
+            self._sendAnswer(b"%f" % (self._temperature,))
+            self._temperature += 1
```

Comment:
Please change to a random value around the setpoint. Otherwise, in the simulated system, the temperature will get very, confusingly, hot after a couple of minutes.